File: CPP/Memory.cpp
#include <iostream>
#include <string>

using namespace std;

class Memory : public Hardware
{
    private:
        int frequency;
		int memorySize;
		string supportsCuda;

    public:

        Memory(int freq, int memo, string supp)
        {
            setFreq(freq);
            setMemo(memo);
            setSupp(supp);
        }
        Memory()
        {

        }

        void setFreq(int freq)
        {
            this->frequency = freq;
        }
        int getFreq()
        {
            return frequency;
        }

        void setMemo(int memo)
        {
            this->memorySize = memo;
        }
        int getMemo()
        {
            return memorySize;
        }

        void setSupp(string supp)
        {
            this->supportsCuda = supp;
        }
        string getSupp()
        {
            return supportsCuda;
        }

};

File: Java/Memory.java
public class Memory extends Hardware{

    private int freq;
    private int memo;
    private String supp;

    public Memory(int freq, int memo, String supp) 
    {
        setFreq(freq);
        setMemo(memo);
        setSupp(supp);
    }

    public Memory() {

    }

    public void setFreq(int frequency) 
    {
        this.freq = frequency;
    }
    public int getFreq() 
    {
        return this.freq;
    }

    public void setMemo(int memorySize) 
    {
        this.memo = memorySize;
    }
    public int getMemo() 
    {
        return this.memo;
    }

    public void setSupp(String supportsCuda) 
    {
        this.supp = supportsCuda;
    }
    public String getSupp() 
    {
        return this.supp;
    }

}